guides/doc-Provisioning_Guide/topics/proc_configuring_iPXE_to_reduce_provisioning_times.adoc

```diff
@@ -0,0 +1,348 @@
+
+. Virtual machine starts
+. iPXE retrieves the network credentials using DHCP
+. iPXE retrieves the HTTP address using DHCP
+. iPXE chainloads the iPXE template from the template {SmartProxy}
+. iPXE loads the kernel and initial RAM disk of the installer
+
+Ensure that the hypervisor that you want to use supports iPXE. The following virtualization hypervisors support iPXE:
+
+* libvirt
+* oVirt
+* RHEV
+
+.Configuring {ProjectName} Server to use iPXE
+
+You can use the default template to configure iPXE booting for hosts.
+If you want to change the default values in the template, clone the template and edit the clone.
+
+.Procedure
+
+To configure {Project} to use iPXE, complete the following steps:
+
+. In the {Project} web UI, navigate to *Hosts* > *Provisioning Templates*, enter `Kickstart default iPXE` and click *Search*.
+. Optional: If you want to change the template, click *Clone*, enter a unique name, and click *Submit*.
+. Click the name of the template you want to use.
+. If you clone the template, you can make changes you require on the *Template* tab.
+. Click the *Association* tab, and select the operating systems that your host uses.
+. Click the *Locations* tab, and add the location where the host resides.
+. Click the *Organizations* tab, and add the organization that the host belongs to.
+. Click *Submit* to save the changes.
+. Navigate to *Hosts* > *Operating systems* and select the operating system of your host.
+. Click the *Templates* tab.
+. From the *iPXE Template* list, select the template you want to use.
+. Click *Submit* to save the changes.
+. Navigate to *Hosts* > *All Hosts*.
+. In the *Hosts* page, select the host that you want to use.
+. Select the *Templates* tab.
+. From the *iPXE template* list, select *Review* and verify that the *Kickstart default iPXE* template is the correct template.
+. To prevent an endless loop of chainbooting iPXE firmware, edit the `/etc/dhcp/dhcpd.conf` file to match the following example. If you use an isolated network, use a {SmartProxyServer} URL with TCP port 8000, instead of {ProjectServer}'s URL.
+.. Locate the following lines in the Bootfile Handoff section of the `/etc/dhcp/dhcpd.conf` file:
++
+----
+} else {
+  filename "pxelinux.0";
+}
+----
+.. Add the following extra `elsif` statement before the else statement:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+elsif exists user-class and option user-class = "iPXE" {
+  filename "http://{foreman-example-com}/unattended/iPXE";
+}
+----
+.. Verify that the 'if' section matches the following example:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+if option architecture = 00:06 {
+  filename "grub2/shim.efi";
+} elsif option architecture = 00:07 {
+  filename "grub2/shim.efi";
+} elsif option architecture = 00:09 {
+  filename "grub2/shim.efi";
+} elsif exists user-class and option user-class = "iPXE" {
+  filename "http://_{foreman-example-com}_/unattended/iPXE";
+} else {
+  filename "pxelinux.0";
+}
```

Comment:
@lzap 

Do I replace this with: 

. To use the iPXE bootstrapping feature for {Project}, configure the `dhcpd.conf` file as follows:
+
----
if exists user-class and option user-class = "iPXE" {
  filename "http://foreman.nat.lan/unattended/iPXE?bootstrap=1";
} elsif option architecture = 00:06 {
  filename "ipxe.efi";
} elsif option architecture = 00:07 {
  filename "ipxe.efi";
} elsif option architecture = 00:09 {
  filename "ipxe.efi";
} else {
  filename "undionly.0";
}
----

Reply:
Yes that appears to be correct.